src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp

```diff
@@ -85,12 +84,66 @@ ep_rt_aot_sample_profiler_write_sampling_event_for_threads (
 {
 }
 
+#if defined (__linux__)
+extern char *__progname;
+#endif
+
 const ep_char8_t *
 ep_rt_aot_entrypoint_assembly_name_get_utf8 (void) 
 {
-    // shipping criteria: no EVENTPIPE-NATIVEAOT-TODO left in the codebase
-    // TODO: Implement EventPipe assembly name - return filename in nativeaot?
-    return reinterpret_cast<const ep_char8_t*>("");
+    // Although PalGetModuleFileName exists, non-windows implementation has issues and we use __progname for Linux.
+    // Cannot use __cpp_threadsafe_static_init feature since it will bring in the C++ runtime and need to use threadsafe way to initialize entrypoint_assembly_name
+    static const ep_char8_t * entrypoint_assembly_name = nullptr;
+    if (entrypoint_assembly_name == nullptr) {
+        const ep_char8_t * entrypoint_assembly_name_local;
+#ifdef HOST_WINDOWS
+        const TCHAR * wszModuleFileName = NULL;
+        if(PalGetModuleFileName(&wszModuleFileName, nullptr) == 0)
+            entrypoint_assembly_name_local = reinterpret_cast<const ep_char8_t*>("");
+        else {
+            const wchar_t* process_name = wcsrchr(wszModuleFileName, '\\');
+            if (process_name != NULL) {
+                const wchar_t* extension = wcschr(process_name + 1, '.');
```

Comment:
nit: update `process_name` instead of having to do `process_name + 1` throughout